File: app/src/main/java/com/manrique/trailerstock/dao/PromocionProductoDao.java
package com.manrique.trailerstock.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import com.manrique.trailerstock.model.PromocionProducto;
import java.util.List;

@Dao
public interface PromocionProductoDao {

    @Insert
    void insertar(PromocionProducto promocionProducto);

    @Query("SELECT * FROM promocion_productos WHERE promocionId = :promocionId")
    List<PromocionProducto> obtenerProductosPorPromocion(int promocionId);

    @Query("DELETE FROM promocion_productos WHERE promocionId = :promocionId")
    void eliminarProductosPorPromocion(int promocionId);

    @Delete
    void eliminar(PromocionProducto promocionProducto);
}

File: app/src/main/java/com/manrique/trailerstock/data/local/dao/PromocionProductoDao.kt
package com.manrique.trailerstock.data.local.dao

import androidx.room.*
import com.manrique.trailerstock.data.local.entities.PromocionProducto

/**
 * DAO para operaciones de PromocionProducto (relación muchos a muchos).
 */
@Dao
interface PromocionProductoDao {
    
    @Query("SELECT * FROM promocion_productos WHERE promocionId = :promocionId")
    suspend fun obtenerProductosPorPromocion(promocionId: Int): List<PromocionProducto>
    
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertar(promocionProducto: PromocionProducto)
    
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertarVarios(promocionProductos: List<PromocionProducto>)
    
    @Query("DELETE FROM promocion_productos WHERE promocionId = :promocionId")
    suspend fun eliminarProductosDePromocion(promocionId: Int)
    
    @Delete
    suspend fun eliminar(promocionProducto: PromocionProducto)
}
